<a href="https://colab.research.google.com/github/ayanbabusona/tensorflow/blob/master/mnist.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

*Install and Initialize* **Tensorflow**

In [0]:
from __future__ import absolute_import,division,print_function,unicode_literals
!pip install tensorflow==2.0.0-beta1

import tensorflow as tf

*Load and prepare* **MNIST** *data*

In [0]:
mnist = tf.keras.datasets.mnist

(x_train,y_train),(x_test,y_test)=mnist.load_data()
x_train,x_test = x_train/255.0,x_test/255.0

Build the sequeential model by training layers. Choose an optimizer and loss by training.

In [0]:
model = tf.keras.models.Sequential([
    tf.keras.layers.Flatten(input_shape=(28,28)),
    tf.keras.layers.Dense(128,activation='relu'),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(10,activation='softmax')
])

model.compile(optimizer='adam',
             loss ='sparse_categorical_crossentropy',
             metrics=['accuracy'])

Train and evaluate the model

In [23]:
model.fit(x_train,y_train,epochs=10)

model.evaluate(x_test,y_test)

W0826 05:36:46.537679 140184916453248 deprecation.py:323] From /usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where


Train on 60000 samples
Epoch 1/10
60000/60000 [==============================] - 7s 110us/sample - loss: 0.3031 - accuracy: 0.9114
Epoch 2/10
60000/60000 [==============================] - 6s 95us/sample - loss: 0.1460 - accuracy: 0.9569
Epoch 3/10
60000/60000 [==============================] - 6s 108us/sample - loss: 0.1071 - accuracy: 0.9676
Epoch 4/10
60000/60000 [==============================] - 6s 94us/sample - loss: 0.0883 - accuracy: 0.9725
Epoch 5/10
60000/60000 [==============================] - 6s 95us/sample - loss: 0.0751 - accuracy: 0.9760
Epoch 6/10
60000/60000 [==============================] - 6s 96us/sample - loss: 0.0660 - accuracy: 0.9786
Epoch 7/10
60000/60000 [==============================] - 6s 96us/sample - loss: 0.0597 - accuracy: 0.9812
Epoch 8/10
60000/60000 [==============================] - 6s 95us/sample - loss: 0.0532 - accuracy: 0.9828
Epoch 9/10
60000/60000 [==============================] - 5s 91us/sample - loss: 0.0481 - accuracy: 0.9843
Epoch 10/10


[0.06669218131987145, 0.9803]

This image classifier is trained to ~98% accuracy on this dataset.